client/my-sites/checkout/checkout-thank-you/transfer-pending/index.tsx

```diff
@@ -0,0 +1,130 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+import * as React from 'react';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import { useDispatch, useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import getAtomicTransfer from 'calypso/state/selectors/get-atomic-transfer';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { transferStates } from 'calypso/state/atomic-transfer/constants';
+import { hideMasterbar, showMasterbar } from 'calypso/state/ui/masterbar-visibility/actions';
+import { useInterval } from '../../../../lib/interval/use-interval';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+// Total time to perform "loading"
+const DURATION_IN_MS = 60000;
+
+// Props type
+interface Props {
+	siteId: number;
+	orderId: number;
+}
+
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+const TransferPending: React.FunctionComponent< Props > = ( props ) => {
+	const { __ } = useI18n();
+	const { siteId, orderId } = props;
+	const dispatch = useDispatch();
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const transfer = useSelector( ( state ) => getAtomicTransfer( state, siteId ) );
+
+	const steps = React.useRef< string[] >( [
+		__( 'Setting up your site' ),
+		__( 'Upgrading infrastructure' ),
+		__( 'Preparing WooCommerce' ),
+	] );
+
+	const totalSteps = steps.current.length;
+
+	const [ currentStep, setCurrentStep ] = React.useState( 0 );
+
+	/**
+	 * Completion progress: 0 <= progress <= 1
+	 */
+	const progress = ( currentStep + 1 ) / totalSteps;
+	const isComplete = progress >= 1;
+
+	useInterval(
+		() => setCurrentStep( ( s ) => s + 1 ),
+		// Enable the interval when progress is incomplete
+		isComplete ? null : DURATION_IN_MS / totalSteps
+	);
+
+	// Force animated progress bar to start at 0
+	const [ hasStarted, setHasStarted ] = React.useState( false );
+	React.useEffect( () => {
+		const id = setTimeout( () => setHasStarted( true ), 750 );
+		return () => clearTimeout( id );
+	}, [] );
+
+	// Hide toolbar while component is mounted
+	React.useEffect( () => {
+		dispatch( hideMasterbar() );
+		return () => {
+			dispatch( showMasterbar() );
+		};
+	}, [ dispatch ] );
+
+	// Redirect based on transfer status
+	React.useEffect( () => {
+		const retryOnError = () => {
```

Comment:
We could use a better error experience here, but this is a copy & paste of how it is now.